src/Agent.Worker/TaskRunner.cs

```diff
@@ -645,10 +645,17 @@ private void PublishTelemetry(Definition taskDefinition, HandlerData handlerData
                 publishTelemetryCmd.Initialize(HostContext);
                 publishTelemetryCmd.ProcessCommand(ExecutionContext, cmd);
             }
-            catch (NullReferenceException ex)
-            {
-                ExecutionContext.Debug($"ExecutionHandler telemetry wasn't published, because one of the variables is null");
-                ExecutionContext.Debug(ex.ToString());
+            catch (Exception ex)            
+            {                    
+                if (ex is FormatException || ex is ArgumentNullException || ex is NullReferenceException)
```

Comment:
nit: You can use `when` ex filter: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/exception-handling-statements#a-when-exception-filter

Reply:
added exception filter, thanks!